### PR TITLE
support for nexus 3.38 fixes #22

### LINKED ISSUE #22: Release for nxrm 3.38.0-01
* **What are you trying to do?**
Upgrade to nxrm 3.38.0-01

* **What feature or behavior is this required for?**
Would it be possible to get a release compatible with nxrm 3.38.0-01?
We've upgraded to this version and found that nxrm wouldn't start because of the existence of a cpan proxy repository

* **How could we solve this issue? (Not knowing is okay!)**

* **Anything else?**

### PR DESCRIPTION
**Pull request by christophvw**
support for nexus 3.38

### CODE REVIEW

File: nexus-repository-cpan/src/main/java/org/sonatype/nexus/repository/cpan/datastore/internal/CpanProxyRecipe.java [hunk 1]
```diff
@@ -89,7 +89,7 @@
 
   private final Provider<ConfigurableViewFacet> viewFacet;
 
-  private final Provider<SearchFacet> searchFacet;
+  private final Provider<ElasticSearchFacet> searchFacet;
```
Comment (bhamail) [inline nexus-repository-cpan/src/main/java/org/sonatype/nexus/repository/cpan/datastore/internal/CpanProxyRecipe.java:92]: see above comment about `org.sonatype.nexus.repository.content.search.SearchFacet`.

**Review by bhamail — CHANGES_REQUESTED** [review]
Looks good to me. One question (see inline) regarding the change to `see above comment about `org.sonatype.nexus.repository.content.search.SearchFacet`. (Happy to be told it is required, but wanted to confirm first).

**Comment by christophvw** [comment]
Now I am stuck :/
Tried to revert the SearchFacet changes but used a different mail address. Now the bots wants me to sign the cla again, but when I click on login to github I get "you already signed the cla"...

File: nexus-repository-cpan/src/main/java/org/sonatype/nexus/repository/cpan/orient/internal/CpanProxyRecipe.groovy [hunk 2]
```diff
@@ -32,7 +32,7 @@ import org.sonatype.nexus.repository.http.PartialFetchHandler
 import org.sonatype.nexus.repository.httpclient.HttpClientFacet
 import org.sonatype.nexus.repository.proxy.ProxyHandler
 import org.sonatype.nexus.repository.purge.PurgeUnusedFacet
-import org.sonatype.nexus.repository.search.ElasticSearchFacet
+import org.sonatype.nexus.repository.search.SearchFacet
```
Comment (bhamail) [inline nexus-repository-cpan/src/main/java/org/sonatype/nexus/repository/cpan/orient/internal/CpanProxyRecipe.groovy:35]: Needs `content` package instead?

File: nexus-repository-cpan/src/main/java/org/sonatype/nexus/repository/cpan/orient/internal/CpanProxyRecipe.groovy [hunk 3]
```diff
@@ -32,7 +32,7 @@ import org.sonatype.nexus.repository.http.PartialFetchHandler
 import org.sonatype.nexus.repository.httpclient.HttpClientFacet
 import org.sonatype.nexus.repository.proxy.ProxyHandler
 import org.sonatype.nexus.repository.purge.PurgeUnusedFacet
-import org.sonatype.nexus.repository.search.SearchFacet
+import org.sonatype.nexus.repository.search.ElasticSearchFacet
```
Comment (bhamail) [inline nexus-repository-cpan/src/main/java/org/sonatype/nexus/repository/cpan/orient/internal/CpanProxyRecipe.groovy:35]: Did you try this instead?:

`import org.sonatype.nexus.repository.content.search.SearchFacet;`

**Review by bhamail — CHANGES_REQUESTED** [review]

File: nexus-repository-cpan/src/main/java/org/sonatype/nexus/repository/cpan/orient/internal/CpanProxyRecipe.groovy (hunk 3 above)
Comment (christophvw) [reply nexus-repository-cpan/src/main/java/org/sonatype/nexus/repository/cpan/orient/internal/CpanProxyRecipe.groovy]: changed

Comment (christophvw) [reply nexus-repository-cpan/src/main/java/org/sonatype/nexus/repository/cpan/orient/internal/CpanProxyRecipe.groovy]: but how should I fix the cla issue now?

Comment (christophvw) [reply nexus-repository-cpan/src/main/java/org/sonatype/nexus/repository/cpan/orient/internal/CpanProxyRecipe.groovy]: ok, added the other mail address now

Comment (bhamail) [reply nexus-repository-cpan/src/main/java/org/sonatype/nexus/repository/cpan/orient/internal/CpanProxyRecipe.groovy]: Noice! Thanks!!

**Review by bhamail — APPROVED** [review]
That's got it.

**Pull request merged**